teuthology/dispatcher/supervisor.py

```diff
@@ -0,0 +1,222 @@
+import logging
+import os
+import subprocess
+import tempfile
+import time
+import yaml
+
+from datetime import datetime
+
+from teuthology import report
+from teuthology import safepath
+from teuthology.config import config as teuth_config
+from teuthology.exceptions import SkipJob
+from teuthology import setup_log_file, install_except_hook
+from teuthology.lock.ops import reimage_many
+from teuthology.misc import get_user
+from teuthology.config import FakeNamespace
+from teuthology.worker import run_with_watchdog, symlink_worker_log
+from teuthology.job_status import get_status
+import teuthology
+from teuthology.nuke import nuke
+
+log = logging.getLogger(__name__)
+start_time = datetime.utcnow()
+restart_file_path = '/tmp/teuthology-restart-workers'
+stop_file_path = '/tmp/teuthology-stop-workers'
+
+
+def main(args):
+
+    verbose = args["--verbose"]
+    archive_dir = args["--archive-dir"]
+    teuth_bin_path = args["--bin-path"]
+    config_fd = int(args["--config-fd"])
+
+    with open(config_fd, 'r') as config_file:
+        config_file.seek(0)
+        job_config = yaml.safe_load(config_file)
+
+    loglevel = logging.INFO
+    if verbose:
+        loglevel = logging.DEBUG
+    log.setLevel(loglevel)
+
+    suite_dir = os.path.join(archive_dir, job_config['name'])
+    if (not os.path.exists(suite_dir)):
+        os.mkdir(suite_dir)
+    log_file_path = os.path.join(suite_dir, 'supervisor.{job_id}'.format(
```

Comment:
Can we put the supervisor log into the job directory, i.e. `f'{run_dir}/{job_id}/supervisor.log'`

Reply:
The only reason I didn't put it there is that we will have unnecessary folders for first and last jobs in a run. If that is not an issue, this makes more sense.

Reply:
Yes... right, the first and last in suite are specific jobs, and they should be handled differently, since they do not start any teuthology job.

Reply:
Done. Removing the unnecessary dirs created at the end of execution in case the job is first or last in suite [here](https://github.com/ceph/teuthology/pull/1546/files#diff-159cd41bfcdb054c0e04dc2650bfa033R93-R96).